runtime/src/system_instruction_processor.rs

```diff
@@ -16,130 +16,89 @@ use solana_sdk::{
 
 use std::collections::HashSet;
 
-fn create_account_with_seed(
-    from: &mut KeyedAccount,
-    to: &mut KeyedAccount,
-    base: &Pubkey,
-    seed: &str,
-    lamports: u64,
-    data_length: u64,
-    program_id: &Pubkey,
-    signers: &HashSet<Pubkey>,
-) -> Result<(), InstructionError> {
-    // `base` is the source of the derived address and must sign for
-    //   rights to the address
-    if !signers.contains(&base) {
-        debug!("CreateAccountWithSeed: must carry signature of `base`");
-        return Err(InstructionError::MissingRequiredSignature);
-    }
-
-    // re-derive the address, must match the `to` account
-    let address = create_address_with_seed(base, seed, program_id)?;
-
-    if to.unsigned_key() != &address {
-        debug!(
-            "CreateAccountWithSeed: invalid argument; derived {} does not match `to` {}",
-            address,
-            to.unsigned_key()
-        );
-        return Err(SystemError::AddressWithSeedMismatch.into());
-    }
-
-    // all of finish_create_account's rules apply
-    finish_create_account(from, to, lamports, data_length, program_id)
+// represents an address that may or may not have been generated
+//  from a seed
+#[derive(PartialEq, Default, Debug)]
+struct Address {
+    address: Pubkey,
```

Comment:
Would now be a good time to start moving away from abusing `Pubkey` for account addresses? That is, another `[u8; 32]` here?

Reply:
I dunno.  This is an internal type and really specific to my this use case...